registry/scripts/genheaders.py

```diff
@@ -0,0 +1,801 @@
+ * itself.
+ *
+ * ALC_APIENTRY is similarly typically only needed for specifying the calling
+ * convention for functions and function pointers declared in the module.
+ * However, some extensions use callbacks that need user code to define
+ * functions with the same calling convention. Currently this is set to use the
+ * platform's default calling convention (that is, it's defined to nothing),
+ * except on Windows where it's defined to __cdecl. Interestingly, capture-less
+ * lambdas seem to generate conversion operators that match function pointers
+ * of any calling convention, but short of that, the user will be responsible
+ * for ensuring callbacks use the cdecl calling convention on Windows and the
+ * default for other OSs.
+ *
+ * Additionally, enums are declared as global inline constexpr ints. This
+ * should generally be fine as long as user code doesn't try to use them in
+ * the preprocessor, which will no longer recognize or expand them to integer
+ * literals. Being global ints also defines them as actual objects stored in
+ * memory, lvalues whose addresses can be taken, instead of as integer literals
+ * or prvalues, which may have subtle implications. An unnamed enum would be
+ * better here, since the enumerators associate a value with a name and don't
+ * become referenceable objects in memory, except that gives the name a new
+ * type (e.g. typeid(ALC_NO_ERROR) != typeid(int)) which could create problems
+ * for type deduction.
+ *
+ * Note that defining AL_LIBTYPE_STATIC, AL_DISABLE_NOEXCEPT, and/or
+ * ALC_NO_PROTOTYPES does still influence the function and function pointer
+ * type declarations, but only when compiling the module. The user-defined
+ * macros have no effect when importing the module.
+ */
+"""
+
+PLATFORM_MODULE_PREAMBLE = """
+#ifndef {ns}_API
+ #if defined(AL_LIBTYPE_STATIC)
+  #define {ns}_API
+ #elif defined(_WIN32)
+  #define {ns}_API __declspec(dllimport)
+ #else
+  #define {ns}_API extern
+ #endif
+#endif
+
+#ifdef _WIN32
+ #define {ns}_APIENTRY __cdecl
+#else
+ #define {ns}_APIENTRY
+#endif
+
+#ifndef AL_DISABLE_NOEXCEPT
+ #define {ns}_API_NOEXCEPT noexcept
+#else
+ #define {ns}_API_NOEXCEPT
+#endif
+
+#define ENUMDCL inline constexpr auto
+
+export module openal.{mod};
+"""
+
+EXT_MODULE_PREAMBLE = """
+#include <array>
+#include <cstddef>
+#include <cstdint>
+
+#ifndef ALC_API
+ #if defined(AL_LIBTYPE_STATIC)
+  #define ALC_API
+ #elif defined(_WIN32)
+  #define ALC_API __declspec(dllimport)
+ #else
+  #define ALC_API extern
+ #endif
+#endif
+
+#ifdef _WIN32
+ #define ALC_APIENTRY __cdecl
+#else
+ #define ALC_APIENTRY
+#endif
+
+#ifndef AL_DISABLE_NOEXCEPT
+ #define ALC_API_NOEXCEPT noexcept
+#else
+ #define ALC_API_NOEXCEPT
+#endif
+
+#ifndef AL_API
+ #define AL_API ALC_API
+#endif
+#define AL_APIENTRY ALC_APIENTRY
+#define AL_API_NOEXCEPT ALC_API_NOEXCEPT
+
+#define ENUMDCL inline constexpr auto
+
+typedef std::int64_t _alsoft_int64_t;
+typedef std::uint64_t _alsoft_uint64_t;
```

Comment:
These need to be moved to under the `export module openal.ext` line. Only preprocessor directives (and comments) can appear between the `module;` and `export module ...` lines.